theia-extensions/viewer-prototype/src/node/trace-server/trace-server-service.ts

```diff
@@ -23,14 +25,19 @@ export class TraceServerServiceImpl implements TraceServerConfigService {
                 throw new Error('the Trace Server is already running on a different port');
             }
         }
+        if (!path) {
+            path = this.getTraceServerPath();
+        }
+        if (!path) {
+            throw new Error('no Trace Server path found');
+        }
         const server = spawn(path, ['-vmargs', `-Dtraceserver.port=${port}`]);
         if (server.pid === undefined) {
             // When pid is undefined it usually means we're about to get an error.
             return new Promise<never>((_, reject) => server.once('error', reject));
         }
         this.server = server;
         this.port = port;
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
I need to refer to `timeout` in the `finally` clause, so I can't `const` it in the try block. Moreover, I need the `any` because the tsconfigs are setup in such a way that both node and dom typings are visible at the same time which makes `clearTimeout` have two fundamentally incompatible signatures... I still cannot build if I don't type it as `any`.